lib/fs.js

```diff
@@ -86,7 +104,7 @@ Object.defineProperty(fs, 'promises', {
       process.emitWarning('The fs.promises API is experimental',
                           'ExperimentalWarning');
     }
-    return promises;
+    return require('internal/fs/promises');
```

Comment:
I suggest to move the require in the `if (warn)` check to only require it once and to use a extra variable.